src/main/java/org/folio/fqm/migration/MigrationUtils.java

```diff
@@ -101,4 +103,113 @@ private static ObjectNode migrateFqlTree(ObjectNode fql, TriConsumer<ObjectNode,
+   *
+   * This is similar to {@link #migrateFql(String, UnaryOperator, TriConsumer)}, but operates on
+   * each value, and provides additional convenience to handle cases of array values, etc.
+   *
+   * @param fqlQuery The root query to migrate
+   * @param versionTransformer A function that takes the current (potentially null) version and
+   *                           returns the new one to be persisted in the query
+   * @param applies If the valueTransformer should be applied to the field (for optimization)
+   * @param valueTransformer Something that takes an incoming field name, value, and a supplier
+   *                         (to get the original fql for warnings), returning either the
+   *                         transformed value, or `null` if it should be removed (the transformer
+   *                         is responsible for handling warnings)
+   */
+  public static String migrateFqlValues(
+    String fqlQuery,
+    UnaryOperator<String> versionTransformer,
+    Predicate<String> applies,
+    ValueTransformer valueTransformer
+  ) {
+    return migrateFql(
+      fqlQuery,
+      versionTransformer,
+      (result, key, value) -> {
+        if (!applies.test(key)) {
+          result.set(key, value); // no-op
+          return;
+        }
+
+        ObjectNode conditions = (ObjectNode) value;
+        ObjectNode newValues = objectMapper.createObjectNode();
+
+        conditions
+          .properties()
+          .forEach(entry -> {
+            if (entry.getValue().isArray()) {
+              ArrayNode node = (ArrayNode) entry.getValue();
+
+              List<JsonNode> transformedValues = new ArrayList<>();
+              node.forEach(element -> {
+                JsonNode transformedValue = transformIfStringOnly(
+                  entry.getValue(),
+                  textValue ->
+                    valueTransformer.apply(
+                      key,
+                      textValue,
+                      () -> objectMapper.createObjectNode().set(entry.getKey(), entry.getValue()).toPrettyString()
+                    )
+                );
+
+                if (transformedValue != null) {
+                  transformedValues.add(transformedValue);
+                }
+              });
+
+              if (!transformedValues.isEmpty()) {
+                newValues.set(entry.getKey(), objectMapper.createArrayNode().addAll(transformedValues));
+              }
+            } else {
+              JsonNode transformedValue = transformIfStringOnly(
+                entry.getValue(),
+                textValue ->
+                  valueTransformer.apply(
+                    key,
+                    textValue,
+                    () -> objectMapper.createObjectNode().set(entry.getKey(), entry.getValue()).toPrettyString()
+                  )
+              );
+
+              if (transformedValue != null) {
+                newValues.set(entry.getKey(), transformedValue);
+              }
+            }
+          });
```

Comment:
this is a hot mess, primarily because we need to consider arrays used for inputs (`$in` and friends), so there's a fair bit that looks the same (but with one changed parameter).

The idea is this utility function can handle iteration through all the json nodes, pulling values out of conditions/arrays/etc, and all migration scripts need to do is tell it how to transform (add timezone, use new value, etc etc)